File: test/before_train/trainset_and_testset/java-py/HumanEval_102_choose_num.py
def choose_num(x: int, y: int) -> int:
    return Problem.chooseNum(x, y)

def check(candidate):
    assert candidate(12, 15) == 14
    assert candidate(13, 12) == -1
    assert candidate(33, 12354) == 12354
    assert candidate(5234, 5233) == -1
    assert candidate(6, 29) == 28
    assert candidate(27, 10) == -1
    assert candidate(7, 7) == -1
    assert candidate(546, 546) == 546

def test_check():
    check(choose_num)

test_check()

File: test/before_train/trainset_and_testset/cpp-java/HumanEval_102_choose_num.java
import java.util.*;
import java.lang.reflect.*;
import java.security.*;
import java.math.*;
import java.io.*;
import java.util.stream.*;
class Problem {
    public static long chooseNum(long x, long y) {
        return choose_num(x, y);
    }
    public static long choose_num(long x, long y) {
        if (x == y) {
            return x;
        }
        if (x % 2 == 0) {
            return x;
        }
        if (y % 2 == 0) {
            return y;
        }
        return y - 1;
    }
    public static void main(String[] args) {
    assert(chooseNum((12l), (15l)) == (14l));
    assert(chooseNum((13l), (12l)) == (-1l));
    assert(chooseNum((33l), (12354l)) == (12354l));
    assert(chooseNum((5234l), (5233l)) == (-1l));
    assert(chooseNum((6l), (29l)) == (28l));
    assert(chooseNum((27l), (10l)) == (-1l));
    assert(chooseNum((7l), (7l)) == (-1l));
    assert(chooseNum((546l), (546l)) == (546l));
    }

}